admin/app/templates/authenticated/sessions/session-id.hbs

```diff
@@ -0,0 +1,8 @@
+<HeaderWithTabs
+  @title='Sessions de certification'
+  @tabs={{this.model.tabs}}
+  @model={{this.model.session}}>
+
+  {{outlet}}
+
+</HeaderWithTabs>
```

Comment:
Qu'apporte ce nouveau component par rapport à un mécanisme d'onglets qu'on retrouve dans les autres applis (exemple en certif) :

```
    <nav class="navbar session-details-controls__navbar-tabs">
      <LinkTo @route="authenticated.sessions.details.parameters" class="navbar-item">
          Détails
      </LinkTo>
      <LinkTo @route="authenticated.sessions.details.certification-candidates" class="navbar-item">
          Candidats {{this.certificationCandidatesCount}}
      </LinkTo>
    </nav>
```

Reply:
Initialement mon idée de faire un component était de pouvoir le réutiliser autre part dans l'app Pix Admin (genre si on crée un nouvel onglet on peut juste appeler le composant et pas recopier tout un bloc de code). 
Après je n'avais pas pensé à reprendre la nav bar de Pix Certif ou Pix Orga... est-ce que je la reprends et l'adapte ?

Reply:
@Krysthalia Désolée de casser le groove, mais pour le moment c'est préférable de faire comme dans PixCertif. La mise en commun de components n'est pas encore possible pour le moment de toute façon, et le bloc existant dans PixCertif est, à mon sens, suffisamment lisible. Après je suis OK pour attendre l'avis d'autres personnes, histoire de pas avoir à changer toutes les deux secondes...

Reply:
Non pas de soucis, c'est mieux je pense qu'on ai le même code entre chaque app, ça simplifiera la mise en place du design system ! 
A voir par contre si c'est possible de caler ce code dans un composant tout de même, parce que sinon ça le duplique au moins 2 fois sur Pix Admin. Je pense que ça serait intéressant d'avoir ce morceau à un seul endroit :)

Reply:
J'attends un 2eme avis sur la question pour être sûre de modifier ça ~

Reply:
Oui bon, après il faut pas partir dans la folie de la factorisation. C'est pas non plus un bout super compliqué ou spécifique. A ce compte là, dès qu'on voit 3 divs imbriqués on en fait un component, dès qu'on voit une div et deux boutons on en fait aussi un component... oO

Reply:
Ember et le front en général c'est nouveau pour moi donc je suis pas sûr d'être pertinent sur le sujet. La couplage ça me parle davantage.

L'avantage d'un bout de code factorisé et générique c'est :
- On gagne du temps quand on a besoin de faire ***exactement*** la même chose
- On ne le modifie qu'à un seul endroit lorsqu'on a besoin que le changement soit répercuté partout

L'inconvénient d'un bout de code factorisé et générique c'est :
- On a tendance à lui ajouté plein de ifs et cas particuliers quand on a ***presque*** besoin de faire la même chose 
- On a du mal à détricoté la factorisation quand on a besoin de faire évoluer le code à certains endroit et pas d'autres.

Du coup il faut faire bien attention, c'est un trade-off : 
- Qu'est-ce que je vais gagner en factorisant ? A quelle fréquence cela va-t-il me faire gagner du temps ? Combien de temps ? Qu'est que je vais perdre en couplage ?
- Qu'est-ce que je vais gagner à ne pas factoriser ? Qu'est ce que ça me permet de ne pas découpler ? Quels risques à avoir cette duplication ?

Au doigt mouillé je dirais que ça serait intéressant d'en faire un composant Pix certif à partir du moment où on aurait 3+ fois ce composant dans Pix certif, pas avant (est-ce déjà le cas ? ces trois composants sont-ils susceptibles d'évoluer de la même façon ?). Faire des composants cross-app me parait risqué. Différents PO / Différents besoins potentiels / Différents designs potentiels (peut-être pas aujourd'hui mais probable à l'avenir).

J'ai pas le recul nécessaire pour avoir un avis plus tranché que ça... 

Sur le sujet : 
- https://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction
- https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)
- Notion de Shared Kernel en DDD

Reply:
Pour le moment ce morceau ne devrait apparaître que 2 fois ! A voir pour l'avenir du coup, je vais reprendre le même morceaux de code que dans pix certif ^^
Merci beaucoup pour vos retours ! :)))

Reply:
Je garde garde quand même de côté le composant dans l'historique de mes commits pour le retrouver facilement le jour où on en aura besoin si ça pose pas de soucis :)